.gitmodules

```diff
@@ -1,18 +1,31 @@
 [submodule "fast_align"]
 	path = 3rd_party/fast_align
 	url = https://github.com/clab/fast_align
+
 [submodule "extract-lex"]
 	path = 3rd_party/extract-lex
 	url = https://github.com/marian-nmt/extract-lex
+
+[submodule "inference-engine/3rd_party/browsermt-marian-dev"]
+	path = inference-engine/3rd_party/browsermt-marian-dev
```

Comment:
We already have this module in the root, can we reuse it somehow? Also, it is possible to have all the 3rd party modules in the root?

Reply:
I actually wrote a long comment about this, and I think it got lost in one of my rebases (since these subtree-merge rebases get complicated).

```
# This is the same dependency and repository as `3rd_party/browsermt-marian-dev` below.
#
# When forking `inference-engine` into to this project, I made an earnest attempt to utilize the preexisting
# `3rd_party/browsermt-marian-dev` submodule within `inference-engine`. Unfortunately, I ran into several roadblocks:
#
#   1) I cannot directly add `3rd_party/browsermt-marian-dev` as a cmake subdirectory because cmake is aware that
#      this path is not a subdirectory of the `inference-engine` project root.
#
#   2) Symbolic links do not appear to work for git submodule direcotires the way that they do for regular directories.
#      Even if the symbolic link had linked correctly, it may have still failed due to the considerations of 1).
#
#   3) I tried using cmake to copy the files from `3rd_party/browsermt-marian-dev` into `inference-engine/3rd_party/browsermt-marian-dev`
#      at build time, which would ensure that there is no duplicate reference to the URL in this file, however the upstream dependency itself
#      has hard-coded expectations that the `.git` directory is only one level up, which appears to work correctly for the way git submodules are
#      configured, but does not work if the files are copied over to a regular directory deeper in the repository's directory tree.
#
# It may be possible to remove `3rd_party/browsermt-marian-dev` to instead use `inference-engine/3rd-party/browsermt-marian-dev` everywhere
# within this repository, but I will leave that for a future commit if there is a need to do so.
[submodule "inference-engine/3rd_party/browsermt-marian-dev"]
path = inference-engine/3rd_party/browsermt-marian-dev
url = https://github.com/browsermt/marian-dev
```

I'll be sure to add the comment back in.